src/dahua.ts

```diff
@@ -28,51 +27,57 @@ class DahuaEvents {
     public RECONNECTING_EVENT_NAME: string = 'reconnecting'
 
     private host:                   string
+    private user:                   string
+    private pass:                   string
+    private requestConfig:          AxiosRequestConfig
 
     constructor(host: string, user: string, pass: string, useHttp: boolean) {
         this.host = host
-        const auth: AxiosBasicCredentials = {
-            username: user,
-            password: pass
-        }
+        this.user = user
+        this.pass = pass
 
-        let keepAliveAgent
-        if(useHttp) {
-            keepAliveAgent = new HttpAgent({
+        const keepAliveAgent: HttpAgent | HttpsAgent = useHttp ?
+            new HttpAgent({
                 keepAlive: this.AGENT_SETTINGS.keepAlive,
                 keepAliveMsecs: this.AGENT_SETTINGS.keepAliveMsecs,
                 maxSockets: this.AGENT_SETTINGS.maxSockets,
                 maxFreeSockets: this.AGENT_SETTINGS.maxFreeSockets,
-            }) 
-        } else {
-            keepAliveAgent = new HttpsAgent({
+            })
+            :
+            new HttpsAgent({
                 keepAlive: this.AGENT_SETTINGS.keepAlive,
                 keepAliveMsecs: this.AGENT_SETTINGS.keepAliveMsecs,
                 maxSockets: this.AGENT_SETTINGS.maxSockets,
                 maxFreeSockets: this.AGENT_SETTINGS.maxFreeSockets,
                 rejectUnauthorized: false,
                 minVersion: "TLSv1"
             })
-        }
 
-        let useSSL = useHttp ? 'http': 'https'
-        const axiosRequestConfig: AxiosRequestConfig ={
-            url: `${useSSL}://${host}${this.EVENTS_URI}`,
+        this.requestConfig = {
+            url: `${useHttp ? 'http': 'https'}://${host}${this.EVENTS_URI}`,
             httpsAgent: keepAliveAgent, 
             httpAgent: keepAliveAgent,
-            auth: auth,
-            headers: this.HEADERS,
             method: 'GET',
-            responseType: 'stream'
+            responseType: 'stream',
+
         }
 
         this.eventEmitter = new EventEmitter()
 
-        this.connect(axiosRequestConfig, 0)
+        this.connect(0)
     }
 
-    private connect = (axiosRequestConfig: AxiosRequestConfig, count: number) => {
-        Axios.request(axiosRequestConfig).then((res: AxiosResponse) => {
+    private connect = (count: number, authHeader?: string) => {
+        Axios.request({
+            ...this.requestConfig, 
+            headers: {
+                'accept': 'multipart/x-mixed-replace',
+                'authorization': authHeader ?
+                    authHeader :
+                    // basic auth
+                    Buffer.from(`${this.user}:${this.pass}`).toString('base64')
+            }

```

Comment:
I'd prefer to keep this request config intact. I'd rather pass around this object than a nullable digest auth header string. Additionally passing around this object will eliminate this addition. Plus the addition of a `requestConfig` class variable will not be needed.

Reply:
OK I've refactored this back to the `const axiosRequestConfig` and the `const auth`.

To workaround the axios auth header issue I had to introduce a bit of code that unset `axiosRequestConfig.auth` if the `authorization` header is set. It seems to work in my local test.

This makes it a non-pure function which is a bit awkward. 

The alternative is an inline spread update like this, not sure if you have a preference
```ts
        Axios.request({
            ...axiosRequestConfig,
            // if custom authorization header is set for digest auth, unset the axios auth config 
            // because Axios will always send basic auth and not send the custom authorization header if "auth" is set
            auth: axiosRequestConfig.headers?.['authorization'] ? undefined : axiosRequestConfig.auth
        }).then((res: AxiosResponse) => {
```

Reply:
> This makes it a non-pure function which is a bit awkward.

Agreed, though I can not think of a better solution. I think this is an acceptable workaround and you provided an excellent clarification comment of why it's needed. (Thank you!)

Reply:
> The alternative is an inline spread update like this, not sure if you have a preference

No preference, thank you for asking!

Reply:
Actually if we just patched it internally it would keep it as a pure function, thoughts on this?

```ts
    private connect = (axiosRequestConfig: AxiosRequestConfig, count: number) => {
        // if custom authorization header is set for digest auth, unset the axios auth config 
        // because Axios will always send basic auth and not send the custom authorization header if "auth" is set
        const authPatchedRequestConfig = {
            ...axiosRequestConfig,
            auth: axiosRequestConfig.headers?.['authorization'] ? undefined : axiosRequestConfig.auth
        }

        Axios.request(authPatchedRequestConfig).then((res: AxiosResponse) => {
```

Reply:
Actually can put that into a separate function

Reply:
Sorry I just realized the non-pure function approach had a bug because after we overwrite `axiosRequestConfig.auth = undefined`, any future auth errors won't work because it won't be able to generate a new digest hash because it's referencing the config

```
 const HA1 = md5(`${axiosRequestConfig.auth?.username}:${realm}:${axiosRequestConfig.auth?.password}`)
```

This was the reason why I had the `user` `pass` on the class separate to the config.

I think with a separate function this will be fine since it doesn't actually touch `axiosRequestConfig.auth`